src/accelerate/checkpointing.py

```diff
@@ -212,23 +214,24 @@ def load_accelerator_state(
         else:
             # Load with torch
             input_model_file = input_dir.joinpath(f"{MODEL_NAME}{ending}.bin")
-            state_dict = torch.load(input_model_file, map_location=map_location)
+            state_dict = load(input_model_file, map_location=map_location, weights_only=True)
```

Comment:
Does this pass for models in general? I wonder because of the example I had posted on slack:

```python
import torch
from torch import nn

print(torch.__version__)  # 2.4.0

class MyModule(nn.Module):
    def __init__(self):
        super().__init__()
        self.dense = nn.Linear(1, 1)

net = MyModule()
torch.save(net, "/tmp/checkpoint.pt")

# uncomment to remove the error
# torch.serialization.add_safe_globals([MyModule, set, nn.Linear])
torch.load("/tmp/checkpoint.pt", weights_only=True)
```

Reply:
no, only state dicts (which is what we save here)